### ISSUE TITLE
在训练过程中验证模型，train loss 不下降

### ISSUE THREAD

**Issue by ShaneTian** [PaddlePaddle/Paddle#25127]
- 版本、环境信息：
    - PaddlePaddle版本：`paddlepaddle-gpu 1.7.0.post107`
    - GPU：`TITAN RTX`、`cudatoolkit 10.0.130`、`cudnn 7.3.1`
    - 系统环境：`CentOS Linux 7.6.1810`、`Python 3.7.4`
- 训练信息
    - 单机单卡，通过 `CUDA_VISIBLE_DEVICES=0` 限制为单卡
    - 显存信息：`3178MiB / 24190MiB`

---

**需求**：在训练的过程中验证模型效果，但数据源不同。

**做法（已简化）**：
```py
model = network()

train_prog = fluid.default_main_program()
train_startup = fluid.default_startup_program()

loss = model.loss
mean_acc = model.mean_acc

val_prog = train_prog.clone(for_test=True)
test_prog = train_prog.clone(for_test=True)

optimizer = fluid.optimizer.Adam(learning_rate=1e-3)
optimizer.minimize(loss)

place = fluid.CUDAPlace(0)
exe = fluid.Executor(place)
exe.run(train_startup)

compiled_train_prog = fluid.CompiledProgram(train_prog).with_data_parallel(loss_name=loss.name)
compiled_val_prog = fluid.CompiledProgram(val_prog).with_data_parallel(share_vars_from=compiled_train_prog)
compiled_test_prog = fluid.CompiledProgram(test_prog).with_data_parallel(share_vars_from=compiled_train_prog)

for train_data in train_reader():
    # ...
    (train_cur_loss, train_cur_acc) = exe.run(program=compiled_train_prog, feed=train_data,
                                              fetch_list=[loss.name, mean_acc.name])
    # ...
    for eval_data in val_reader():
        eval_cur_acc, = exe.run(program=compiled_val_prog, feed=eval_data,
                                fetch_list=[mean_acc.name])
    # ...
```
全部训练代码请见 https://github.com/ShaneTian/Induction-Networks/blob/master/train.py

**问题**：在第一次 val 之前，`train_cur_loss` 是不断下降的，但是在做完一次 val 之后，再循环 train 时，loss 不会再下降了

**Comment by MRXLT** [comment]
你好，loss不下降是指的loss值保持不变吗，还是loss值没有第一次val之前的值低但是还是在变化的

**Comment by ShaneTian** [comment]
> 你好，loss不下降是指的loss值保持不变吗，还是loss值没有第一次val之前的值低但是还是在变化的

你好。是在第一次 val 之后保持不变了，维持在一个固定的数值，可见 https://aistudio.baidu.com/aistudio/projectdetail/296000

**Comment by ShaneTian** [comment]
请问是什么原因呢？我又检查了一遍代码，还是没找到原因。。 @MRXLT

**Comment by MRXLT** [comment]
你好，提供的aistudio中的代码第一次val之后loss仍然是有变化的
![image](https://user-images.githubusercontent.com/16594411/86198074-65241a80-bb89-11ea-9554-faa21b3ff8b9.png)

**Comment by ShaneTian** [comment]
> 你好，提供的aistudio中的代码第一次val之后loss仍然是有变化的
> ![image](https://user-images.githubusercontent.com/16594411/86198074-65241a80-bb89-11ea-9554-faa21b3ff8b9.png)

请问您 fork 代码之后直接运行的吗？没有做任何修改吗？
我又试了好几次。。。还是第一次 val 之后是 0.25。。。
而且 loss 的量级感觉和您的不一致，大约在 1/4 附近。

期待您的再次回复，谢谢！
@MRXLT 

![image](https://user-images.githubusercontent.com/42370681/86250241-ea87e900-bbe2-11ea-8448-29c5cd16d02f.png)
![image](https://user-images.githubusercontent.com/42370681/86250288-fa9fc880-bbe2-11ea-9fda-2665cf75ca3f.png)
![image](https://user-images.githubusercontent.com/42370681/86250316-01c6d680-bbe3-11ea-801f-86ac9b2df666.png)

**Comment by ShaneTian** [comment]
已解决，谢谢！
